### ISSUE TITLE
Potentially a concurrency issue while installing packages

### ISSUE THREAD

**Issue by rootcss** [python-poetry/poetry#4568]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

## Issue
While running the `poetry install` command, it randomly fails (not everytime) with errors like:
```
ERROR: Could not install packages due to an EnvironmentError: [Errno 2] No such file or directory: '/tmp/build_8c19dde7/.heroku/python/lib/python3.8/site-packages/~dna-3.2.dist-info'
```
This doesn't happen for a specific package, but rather for any random package. @finswimmer has responded to it in the following link: https://githubmemory.com/repo/python-poetry/poetry/issues/4297 but this is still happening. It potentially looks like a concurrency issues while installing the packages as mentioned by [cfra](https://githubmemory.com/@cfra)

**Comment by uSpike** [comment]
Just want to add that I'm seeing this issue randomly across CI and multiple `poetry install` executions.  Trying to craft a reproduction case...

**Comment by jedie** [comment]
> I'm seeing this issue randomly across CI and multiple `poetry install` executions

Think i have the same problem in Github Actions, because of caching e.g.:

* https://github.com/boxine/bx_django_utils/actions/runs/3096141479/jobs/5011337500
* https://github.com/boxine/bx_django_utils/actions/runs/3088280854/jobs/4994569633

Think i have to split the cache.

**Comment by neersighted** [comment]
Should be fixed by #6186 if this is indeed concurrent writes to the cache -- there is not enough detail to tell if it could be something else, so I am closing this barring more information (especially a reproduction)
